DBCON-229: Update DB's Doc on how Pooling works within Stored Procedures (#1079)
mulesoft/docs-connectors: db/1.8/modules/ROOT/pages/database-documentation.adoc

=== For Configurations
* <<config>>

=== Working with Pooling Profiles
When working with pooling profiles and the *Select* operation, the connection remains open until the flow execution ends or the content of the streams are consumed completely, or if the connection is the transaction key. This behavior occurs because the resultset the operation generates can have a stream or be part of an ongoing transaction.

> Starting with Database Connector 1.8.3, the connections on the *Select* operation are released if they are not part of a stream or transaction.

Reviewer: Remove this line, doesn't apply to Select operation

=== For Configurations
* <<config>>

=== Working with Pooling Profiles

> The *Stored procedure* operation closes the connections it receives when the flow that contains the operation completes. This behavior occurs because the resultset the operation generates can have a stream or be part of an ongoing transaction. Thereby, Mule runtime engine never knows when the flow is going to consume the resultset in its entirety, nor when the connection will be needed.

Reviewer: I will make this generic to all operations that do streaming.
Also I would say that the connection remains open until the flow execution ends OR the content of the streams are consumed completely or if the conection is the transaction key, will be closed once the transaction ends.

I would rmeove" Thereby, Mule runtime engine never knows when the flow is going to consume the resultset in its entirety, nor when the connection will be needed."

Reviewer 2: @estebanwasinger @julodiaz Could you confirm the DB operations that do streaming? so I can make this information generic and add it to all the mentioned operations. Thanks.

Author: The intention of the comment was to document the particular situation that SP present when closing resultSets, as it came up in the following [SE](https://www.mulesoft.org/jira/browse/SE-17811) (see comments).

Reviewer 2: @estebanwasinger let me know your final comments on this, so I can update it accordingly. Thanks

> The Stored procedure operation closes the connections it receives once the flow containing the operation completes. This behavior occurs because the resultset the operation generates can have a stream or be part of an ongoing transaction. Thereby, Mule runtime engine never knows when the flow is going to consume the resultset in its entirety nor when the connection is going to be needed.

Reviewer 3: ```suggestion
The *Stored procedure* operation closes the connections it receives when the flow that contains the operation completes. This behavior occurs because the resultset the operation generates can have a stream or be part of an ongoing transaction. Thereby, Mule runtime engine never knows when the flow is going to consume the resultset in its entirety, nor when the connection will be needed.
```

> Starting on Database Connector v1.8.3 the connections on the Stored procedure operation are released if they are not part of a stream nor a transaction.

Reviewer 3: ```suggestion
Starting with Database Connector 1.8.3, the connections on the *Stored procedure* operation are released if they are not part of a stream or transaction.
```

> When executing the Stored procedure operation with a pooling profile consider that because connections do not release until the flow completes, the number of available connections should be equal or greater than the ones that are going to be used.

Reviewer 3: ```suggestion
When executing the *Stored procedure* operation with a pooling profile, consider that because connections do not release until the flow completes, the number of available connections must be equal to, or greater than, the connections that will be used.
```

> When executing the *Stored procedure* operation with a pooling profile, consider that because connections do not release until the flow completes, the number of available connections must be equal to, or greater than, the connections that will be used.

Reviewer: this is not exclusive for Stored Procedures, is any DB operation basically
